File: MediumTruck.java
public class MediumTruck extends Truck {
    public static final int MAX_VOLUME = 2000;
    public static final int MAX_WEIGHT = 4000;

    public MediumTruck() {
        super();
    }

    @Override
    public void addPack(Pack pack) {
        if (totalVolume() + pack.getVolume() <= MAX_VOLUME &&
                totalWeight() + pack.getWeight() <= MAX_WEIGHT) {
            packages.add(pack);
        }
    }

    @Override
    public int getMaxWeight() {
        return MAX_WEIGHT;
    }

    @Override
    public int getMaxVolume() {
        return MAX_VOLUME;
    }

    @Override
    public int totalVolume() {
        int total = 0;
        for (Pack pack : packages) {
            total += pack.getVolume();
        }
        return total;
    }

    @Override
    public int totalWeight() {
        int total = 0;
        for (Pack pack : packages) {
            total += pack.getWeight();
        }
        return total;
    }
}

File: C#/PackageDeliverySystem/MediumTruck.cs
namespace PackageDeliverySystem
{
    class MediumTruck : Truck
    {
        public static int MEDIUM_MAX_VOLUME = 2000;
        public static int MEDIUM_MAX_WEIGHT = 4000;

        public MediumTruck() : base() { }

        public override void addPack(Pack pack)
        {
            if (totalVolume() + pack.Volume <= MEDIUM_MAX_VOLUME &&
                totalWeight() + pack.Weight <= MEDIUM_MAX_WEIGHT)
            {
                base.addPack(pack);
            }      
        }

        public override int getMaxWeight()
        {
            return MEDIUM_MAX_WEIGHT;
        }

        public override int getMaxVolume()
        {
            return MEDIUM_MAX_VOLUME;
        }

    }
}
